clients/src/main/java/org/apache/kafka/clients/consumer/internals/OffsetAndTimestampInternal.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients.consumer.internals;
+
+import org.apache.kafka.clients.consumer.OffsetAndTimestamp;
+
+import java.util.Optional;
+
+/**
+ * Internal representation of {@link OffsetAndTimestamp}.
```

Comment:
I would add : Internal representation of {@link OffsetAndTimestamp} **that allows negative offsets**

Reply:
Timestamps I assume.

Reply:
it's actually both! he he, so let's maybe add _negative offsets and timestamps_

Reply:
I think the problem is negative timestamp in the response causing `org.apache.kafka.common.KafkaException: java.lang.IllegalArgumentException: Invalid negative timestamp`.  More specifically is this part that was complaining:
```
if (timestamp < 0)
            throw new IllegalArgumentException("Invalid negative timestamp");
```

Reply:
yes, agree that's the failure we noticed on the sys tests, but conceptually we're creating a new `OffsetAndTimestampInternal` class that is the same as the existing `OffsetAndTimestamp`, with the only difference that the former does not throw on negative offsets or negative timestamps, right? so for the class doc makes sense to mention it.